app/components/polaris/page_actions_component.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Polaris
+  class PageActionsComponent < Polaris::NewComponent
+    renders_one :primary_action, -> (primary: true, **system_arguments) do
```

Comment:
@dan-gamble it seems like we don't need wrappers for complex actions. We can pass everything directly to buttons.